101-webapp-linux-sonarqube-postgresql/azuredeploy.json

```diff
@@ -0,0 +1,219 @@
+      "allowedValues": [
+        50,
+        100
+      ],
+      "defaultValue": 100,
+      "metadata": {
+        "description": "Azure database for PostgreSQL pricing tier"
+      }
+    },
+    "databaseSkuName": {
+      "type": "string",
+      "allowedValues": [
+        "PostgreSQLB100",
+        "PostgreSQLB50"
+      ],
+      "defaultValue": "PostgreSQLB100",
+      "metadata": {
+        "description": "Azure database for PostgreSQL sku name : PostgreSQLB100 (Basic 100 DTU tier) and PostgreSQLB50 (Basic 50 DTU tier)"
+      }
+    },
+    "databaseSkuSizeMB": {
+      "type": "int",
+      "allowedValues": [
+        102400,
+        51200
+      ],
+      "defaultValue": 51200,
+      "metadata": {
+        "description": "Azure database for PostgreSQL Sku Size "
+      }
+    },
+    "databaseSkuTier": {
+      "type": "string",
+      "allowedValues": [
+        "Basic"
+      ],
+      "defaultValue": "Basic",
+      "metadata": {
+        "description": "Azure database for PostgreSQL pricing tier"
+      }
+    },
+    "postgresqlVersion": {
+      "type": "string",
+      "allowedValues": [
+        "9.5",
+        "9.6"
+      ],
+      "defaultValue": "9.6",
+      "metadata": {
+        "description": "PostgreSQL version"
+      }
+    }
+  },
+  "variables": {
+    "databaseName": "[concat(parameters('siteName'), 'database')]",
+    "serverName": "[concat(parameters('siteName'), 'pgserver')]",
+    "jdbcSonarUserName": "[concat(parameters('administratorLogin'),'@',variables('serverName'))]",
+    "hostingPlanName": "[concat(parameters('siteName'), 'serviceplan')]"
+
+  },
+  "resources": [
+    {
+      "comments":"This is the Linux web app with the Sonarquebe Docker image",
+      "type": "Microsoft.Web/sites",
+      "name": "[parameters('siteName')]",
+      "properties": {
+        "siteConfig": {
+          "linuxFxVersion": "DOCKER|SONARQUBE"
+        },
+        "name": "[parameters('siteName')]",
+        "serverFarmId": "[variables('hostingPlanName')]",
+        "hostingEnvironment": ""
+      },
+      "apiVersion": "2016-03-01",
+      "location": "[resourceGroup().location]",
+      "dependsOn": [
+        "[variables('hostingPlanName')]",
+        "[variables('databaseName')]"
+      ],
+      "resources": [
+        {
+          "name": "appsettings",
+          "type": "config",
+          "apiVersion": "2015-08-01",
+          "dependsOn": [
+            "[resourceId('Microsoft.Web/sites', parameters('siteName'))]"
+          ],
+          "tags": {
+            "displayName": "SonarappSettings"
+          },
+          "comments": "This are the generated settings for the Sonar jdbc connectionstring",
+          "properties": {
+            "SONARQUBE_JDBC_URL": "[concat('jdbc:postgresql://', variables('serverName') ,'.postgres.database.azure.com:5432/',variables('databaseName'),'?user=', variables('jdbcSonarUserName'), '&password=',parameters('administratorLoginPassword'), '&ssl=true')]",
```

Comment:
same comment as above - use reference if possible

Reply:
Couldn't find function :( but will continue looking in future updates

Reply:
try: reference(resourceId('Microsoft.DBforPostgreSQL/servers',variables('serverName'))).fullyQualifiedDomainName - should work on both mysql and postgre